packages/remix/plugins/component-testing/index.ts

```diff
@@ -0,0 +1,86 @@
+import { nxBaseCypressPreset } from '@nx/cypress/plugins/cypress-preset';
+import { joinPathFragments, workspaceRoot } from '@nx/devkit';
+
+import { existsSync } from 'fs';
+import { dirname, join } from 'path';
+
+type ViteDevServer = {
+  framework: 'react';
+  bundler: 'vite';
+  viteConfig?: any;
+};
+
+/**
+ * Remix nx preset for Cypress Component Testing
+ *
+ * This preset contains the base configuration
+ * for your component tests that nx recommends.
+ * including a devServer that supports nx workspaces.
+ * you can easily extend this within your cypress config via spreading the preset
+ * @example
+ * export default defineConfig({
+ *   component: {
+ *     ...nxComponentTestingPreset(__dirname)
+ *     // add your own config here
+ *   }
+ * })
+ *
+ * @param pathToConfig will be used for loading project options and to construct the output paths for videos and screenshots
+ */
+export function nxComponentTestingPreset(pathToConfig: string): {
+  specPattern: string;
+  devServer: ViteDevServer;
+  videosFolder: string;
+  screenshotsFolder: string;
+  video: boolean;
+  chromeWebSecurity: boolean;
+} {
+  const normalizedProjectRootPath = ['.ts', '.js'].some((ext) =>
+    pathToConfig.endsWith(ext)
+  )
+    ? pathToConfig
+    : dirname(pathToConfig);
+
+  return {
+    ...nxBaseCypressPreset(pathToConfig),
+    specPattern: 'src/**/*.cy.{js,jsx,ts,tsx}',
+    devServer: {
+      ...({ framework: 'react', bundler: 'vite' } as const),
+      viteConfig: async () => {
+        const viteConfigPath = findViteConfig(normalizedProjectRootPath);
+
+        const { mergeConfig, loadConfigFromFile, searchForWorkspaceRoot } =
+          (await import('vite')) as typeof import('vite');
```

Comment:
`import(...)` should already have the right type, why the casting?

Reply:
I didn't notice this. It was copied from the React plugin in Nx. We should probably update there too